rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/dap/breakpoint/BreakpointsCollection.java

```diff
@@ -0,0 +1,45 @@
+package org.rascalmpl.vscode.lsp.dap.breakpoint;
+
+import io.usethesource.vallang.ISourceLocation;
+import org.eclipse.lsp4j.debug.Source;
+import org.rascalmpl.debug.DebugHandler;
+import org.rascalmpl.debug.DebugMessageFactory;
+
+import java.util.HashMap;
+
+/**
+    Store breakpoints set by user
+ **/
+
+public class BreakpointsCollection {
+    private final HashMap<String, HashMap<ISourceLocation, BreakpointInfo>> breakpoints;
```

Comment:
I think it should be a `ConcurrentHashMap`, since we can have multiple request happening in parallel.